jolt-core/src/jolt/instruction/mod.rs

```diff
@@ -9,6 +9,22 @@ pub trait JoltInstruction {
   fn g_poly_degree(&self, C: usize) -> usize;
   fn subtables<F: PrimeField>(&self, C: usize) -> Vec<Box<dyn LassoSubtable<F>>>;
   fn to_indices(&self, C: usize, log_M: usize) -> Vec<usize>;
+  fn lookup_entry<F: PrimeField>(&self, C: usize, M: usize) -> F {
+    let log_M = ark_std::log2(M) as usize;
+
+    let subtable_lookup_indices = self.to_indices(C, ark_std::log2(M) as usize);
+
+    let subtable_lookup_values: Vec<F> = self.subtables::<F>(C)
+      .into_iter()
+      .flat_map(|subtable| {
+          subtable_lookup_indices.iter().map(move |&lookup_index| {
```

Comment:
I believe you can do `.into_iter()` here to avoid the move.

Reply:
(Blaming copilot on that one). 

But yeah, I tried something else that seems to work without having to use move or clone. The tests pass so lemme know if it looks okay.